src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -211,6 +211,9 @@ private static ExternalProjectReference GetExternalProject(PackageSpec rootProje
             // Standard properties
             restoreArgs.ApplyStandardProperties(request);
 
+            // Apply AllowInsecureConnections attribute to RestoreRequest.Project.RestoreMetadata.Sources
```

Comment:
This comment doesn't help me, and in fact makes me wonder why `RestoreRequest` is mentioned. Can it either be elaborated as to why `RestoreRequest` is important, or just delete it?

Also, if you do feel a comment is needed, consider adding a comment-block to the method name, that way you can reference the attribute directly, something like `/// Will apply <see cref="sourceRepository.PackageSource.AllowInsecureConnections"> attribute to <see cref="RestoreRequest.Project.RestoreMetadata.Sources"> for sources in settings in the <paramref name="projectDgSpec" />`.
```suggestion
```

Reply:
Thanks for the question! The `RestoreRequest` is important as we check if the source is http at [here](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs#L242-L246). We need to check additional `source.AllowInsecureConnections`, but the `RestoreRequest.Project.RestoreMetadata.Sources` is not updated to reflect the value in settings. So we have to either updated the `AllowInsecureConnections` flag for `RestoreRequest.Project.RestoreMetadata.Sources`, or updated the whole `RestoreRequest.Project.RestoreMetadata.Sources`.

Reply:
This comment and line of code was deleted, so resolving.